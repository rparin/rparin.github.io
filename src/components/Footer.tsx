import Image from "next/image";
import extLink from "@icons/external-link.svg";

export default function Footer() {
  return (
    <footer className="text-center text-white bg-black py-3 md:py-5">
      <div className="flex flex-row text-center justify-center">
        <p>Built using React and Tailwind.</p>
        <a
          href="https://github.com/rparin/rparin.github.io"
          target="_blank"
          rel="noopener noreferrer">
          <Image
            className="max-w-[1.5vh] invert w-auto h-3/5"
            src={extLink}
            alt="External link to source code"
          />
        </a>
      </div>
      <p>©️ 2023. Designed by Ralph Parin.</p>
    </footer>
  );
}
