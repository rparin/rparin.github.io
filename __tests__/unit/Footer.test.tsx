import { render } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import Footer from "@/components/Footer";

describe("testing Footer Component Accessibility", () => {
  expect.extend(toHaveNoViolations);
  test("it should have no accessibility violations", async () => {
    const { container } = render(<Footer />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

test("it renders component unchanged", () => {
  const { container } = render(<Footer />);
  expect(container).toMatchSnapshot();
});
