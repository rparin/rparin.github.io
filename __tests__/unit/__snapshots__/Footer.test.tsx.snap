// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders component unchanged 1`] = `
<div>
  <footer
    class="bg-primary py-3 text-center text-text md:py-3"
  >
    <div
      class="flex flex-row justify-center text-center"
    >
      <p
        class="text-sm"
      >
        Built using React and Tailwind.
      </p>
      <a
        aria-label="Github, Portfolio Website"
        href="https://github.com/rparin/rparin.github.io"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          class="h-3/5 w-auto max-w-[1.5vh]"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
          />
          <polyline
            points="15 3 21 3 21 9"
          />
          <line
            x1="10"
            x2="21"
            y1="14"
            y2="3"
          />
        </svg>
      </a>
    </div>
    <p
      class="text-sm"
    >
      ©️ 2024. Designed and Developed by Ralph Parin.
    </p>
  </footer>
</div>
`;
